sdk/cognitiveservices/azopenai/custom_client.go

```diff
@@ -239,7 +239,21 @@ func (client *Client) newError(resp *http.Response) error {
 }
 
 type clientData struct {
-	endpoint     string
-	baseEndpoint string
-	azure        bool
+	endpoint string
+	azure    bool
+}
+
+func getDeploymentID[T ChatCompletionsOptions | CompletionsOptions | EmbeddingsOptions | ImageGenerationOptions](v T) string {
```

Comment:
This is a clever way of accomplishing this!

Reply:
I finally got to use generics :)